pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -73,9 +79,13 @@ type ResolvedPipelineRunTask struct {
 	ResolvedConditionChecks TaskConditionCheckState // Could also be a TaskRun or maybe just a Pod?
 }
 
-// PipelineRunState is a slice of ResolvedPipelineRunTasks the represents the current execution
+// PipelineRunState is the combination of the Pipeline dag and
+// a slice of ResolvedPipelineRunTasks the represents the current execution
 // state of the PipelineRun.
-type PipelineRunState []*ResolvedPipelineRunTask
+type PipelineRunState struct {
+	Dag		*dag.Graph
```

Comment:
@afrittoli what do you think about adding a state matrix here in `PipelineRunState` which will help simplify `GetPipelineConditionStatus`?

I had implemented this some time back adding run state matrix in the `PipelineRunStatus`
https://github.com/tektoncd/pipeline/pull/2821/files#diff-4e301064fef70f3c0db810267c0959f2R85 
But now, it would fit well in this data structure.

Reply:
@pritidesai the link points to your comment on this PR, perhaps you meant to share a link to a different PR?

Reply:
😲  sorry @afrittoli have updated the comment ⬆️

Reply:
Thanks!